docs/src/test.md

```diff
@@ -45,3 +45,72 @@ pytest -svv forge/test/mlir/test_ops.py::test_add
+```
+
+Full list of supported query criteria
+
+| Parameter          | Description                                                                                |
+| ------------------ | ------------------------------------------------------------------------------------------ |
+| OPERATORS          | List of operators                                                                          |
+
+To run single test based on a test id use `test_single` method with `TEST_ID` parameter:
+
+```sh
+TEST_ID='ge-FROM_HOST-None-(1, 2, 3, 4)-Float16_b-HiFi4' pytest -svv forge/test/operators/pytorch/test_all.py::test_single
+```
+
+| Parameter          | Description                                                                                |
+| ------------------ | ------------------------------------------------------------------------------------------ |
+| TEST_ID            | Id of a test containing test parameters                                                    |
+
+To see examples of all available tests use `test_unique` method:
+
+```sh
+pytest -svv forge/test/operators/pytorch/test_all.py::test_unique --collect-only
+```
+
+Full list of supported query criteria
+
+| Parameter          | Description                                                                                |
+| ------------------ | ------------------------------------------------------------------------------------------ |
+| OPERATORS          | List of operators                                                                          |
+
+To query subset of test plan based on a query criteria use `test_query` method:
+
+```sh
+OPERATORS=add,div FILTERS=HAS_DATA_FORMAT,QUICK DEV_DATA_FORMATS=Float16_b,Int8 MATH_FIDELITIES=HiFi4,HiFi3 RANGE=5 pytest -svv forge/test/operators/pytorch/test_all.py::test_query --collect-only
+```
+
+```sh
+OPERATORS=add,div FAILING_REASONS=DATA_MISMATCH,UNSUPPORTED_DATA_FORMAT SKIP_REASONS=FATAL_ERROR RANGE=5 pytest -svv forge/test/operators/pytorch/test_all.py::test_query --collect-only
+```
+
+```sh
+FAILING_REASONS=NOT_IMPLEMENTED INPUT_SOURCES=FROM_HOST pytest -svv forge/test/operators/pytorch/test_all.py::test_query --collect-only
+```
+
+Full list of supported query criteria
+
+| Parameter             | Description                                                                                   |
+| --------------------- | --------------------------------------------------------------------------------------------- |
+| OPERATORS             | List of operators                                                                             |
+| FILTERS               | List of labmda filters. See `test/operators/pytorch/test_all.py` for all values               |
+| INPUT_SOURCES         | Supported values: FROM_ANOTHER_OP, FROM_HOST, FROM_DRAM_QUEUE, CONST_EVAL_PASS                |
+| INPUT_SHAPES          | List of input shapes to filter                                                                |
```

Comment:
Do we have limited shapes available based on which we can filter? If yes, let's specify example file path where more details can be found.

Reply:
For filtering tests from test plan shapes are limited.

Live documentation of all supported shapes is now available via `print_query_docs` function from `test_commands.sh` helper scripts.

Reply:
Documentation updated to include `print_query_docs`